lib/Scvmm/miq_scvmm_vm_ssa_info.rb

```diff
@@ -0,0 +1,67 @@
+
+require 'miq_winrm'
+require 'miq_scvmm_parse_powershell'
+
+class MiqScvmmVmSSAInfo
+  attr_reader :vhd, :hostname, :checkpoints, :connection, :vhd_type
+  def initialize(provider, user, pass, port = nil)
+    @checkpoints = []
+    @vhd         = nil
+    @hostname    = nil
+    @vhd_type    = nil
+    @winrm       = MiqWinRM.new
+    winrmport    = port.nil? ? 5985 : port
+    options      = {:port     => winrmport,
+                    :user     => user,
+                    :pass     => pass,
+                    :hostname => provider
+                   }
+
+    @connection  = @winrm.connect(options)
+  end
+
+  def vm_host(vm_name)
+    host_script = <<-host_eol
+    Get-SCVirtualMachine -VMMServer localhost -Name "#{vm_name}" | \
+      Select-Object -ExpandProperty Hostname
+    host_eol
+
+    # TODO: Examine this code for multiple hard drives.
+    @hostname = parse_single_powershell_value(@winrm.run_powershell_script(host_script))
+  end
+
+  def vm_harddisks(vm_name)
+    location_script = <<-location_eol
+    Get-SCVirtualHardDisk -VMMServer localhost -VM "#{vm_name}" | \
+     Select-Object -ExpandProperty Location
+    location_eol
+
+    @vhd_type ||= vm_vhdtype(vm_name)
+    if @vhd_type == "DynamicallyExpanding"
+      @vhd = parse_single_powershell_value(@winrm.run_powershell_script(location_script))
+    elsif @vhd_type == "Differencing"
+      # TODO: return both the Parent Disk and the Checkpoint Disk
+      raise "Currently unsupported VHD Type #{@vhd_type}"
+    else
+      raise "Currently unsupported VHD Type #{@vhd_type}"
+    end
+  end
+
+  def vm_vhdtype(vm_name)
+    vhdtype_script = <<-vhdtype_eol
+    Get-SCVirtualHardDisk -VMMServer localhost -VM "#{vm_name}" | \
+      Select-Object -ExpandProperty VHDType
```

Comment:
@bronaghs - ExpandProperty isn't necessarily *required* but it does make parsing the output eminently easier -
Without ExpandProperty the output is a blank line, the field name, a line of dashes under the field name, the field value, and another blank line.
With ExpandProperty, the output is only the field value.

I remember reading the import-module discussion as well without ever personally experiencing it.  I can certainly add the line - the only thing I can see happening is it slowing down each request slightly more.